### PR TITLE
Implement scheduled event covers

### PR DESCRIPTION
**Pull request by TAG-Epic**
## Summary

This implements scheduled events covers and fixes a bug in fetch_scheduled_event.
Docs are found at https://github.com/Aiko-IT-Systems/Discord-API-Docs/pull/13

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by TAG-Epic** [comment]
Documentation https://github.com/discord/discord-api-docs/pull/4380

File: nextcord/scheduled_events.py [hunk 1]
```diff
@@ -204,8 +204,10 @@ class ScheduledEvent(Hashable):
         The scheduled start time for the event.
     user_count: :class:`int`
         An approximate count of the 'interested' users.
+    cover: :class:`Asset`
+        The event cover image
     """
-    __slots__: Tuple[str] = (
+    __slots__: Tuple[str,...] = (
```
Comment (Gugu7264) [inline nextcord/scheduled_events.py:210]: Why ellipsis in type hint?

File: nextcord/scheduled_events.py [hunk 2]
```diff
@@ -220,6 +222,7 @@ class ScheduledEvent(Hashable):
         'user_count',
         '_state',
         '_users',
+        "cover"
```
Comment (Gugu7264) [inline nextcord/scheduled_events.py:225]: Should use single quotes 👀

**Review by Gugu7264 — CHANGES_REQUESTED** [review]
Not tested

**Comment by Gugu7264** [comment]
Tested, works correctly

File: nextcord/scheduled_events.py (hunk 1 above)
Comment (TAG-Epic) [reply nextcord/scheduled_events.py:210]: Means there are more than one of that type.

File: nextcord/scheduled_events.py (hunk 2 above)
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:225]:
```suggestion
        'cover'
```

File: nextcord/scheduled_events.py [hunk 3]
```diff
@@ -250,6 +253,11 @@ def _update(self, data: ScheduledEventPayload) -> None:
         self.channel_id: Optional[int] = data.get('channel_id')
         self._users: Dict[int, ScheduledEventUser] = {}
         self._update_users(data.get('users', []))
+        
+        if cover := data.get("image"):
+            self.cover: Optional[Asset] = Asset._from_scheduled_event_cover(self._state, self.id, cover)
```
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:258]: Should this be renamed to image as thats what they are called in the API?

File: nextcord/scheduled_events.py [hunk 4]
```diff
@@ -204,8 +204,10 @@ class ScheduledEvent(Hashable):
         The scheduled start time for the event.
     user_count: :class:`int`
         An approximate count of the 'interested' users.
+    cover: :class:`Asset`
+        The event cover image
```
Comment (ooliver1) [inline nextcord/scheduled_events.py:208]:
```suggestion
        The event cover image.
```

File: nextcord/scheduled_events.py (hunk 1 above)
Comment (ooliver1) [inline nextcord/scheduled_events.py:210]:
```suggestion
    __slots__: Tuple[str, ...] = (
```

File: nextcord/scheduled_events.py [hunk 5]
```diff
@@ -220,6 +222,7 @@ class ScheduledEvent(Hashable):
         'user_count',
         '_state',
         '_users',
+        'cover'
```
Comment (ooliver1) [inline nextcord/scheduled_events.py:225]:
```suggestion
        'cover',
```

File: nextcord/scheduled_events.py [hunk 6]
```diff
@@ -250,6 +253,11 @@ def _update(self, data: ScheduledEventPayload) -> None:
         self.channel_id: Optional[int] = data.get('channel_id')
         self._users: Dict[int, ScheduledEventUser] = {}
         self._update_users(data.get('users', []))
+        
+        if cover := data.get("image"):
+            self.cover: Optional[Asset] = Asset._from_scheduled_event_cover(self._state, self.id, cover)
+        else:
+            self.cover: Optional[int] = None
```
Comment (ooliver1) [inline nextcord/scheduled_events.py:260]:
```suggestion
            self.cover: Optional[Asset] = None
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]
tested, however very minor nitpicks

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**